### PR TITLE
add footer-component

### PR DESCRIPTION
**Pull request by Saminsfw**

### CODE REVIEW

File: app/components/FooterMenu/FooterMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      <ul className={`${menuClasses} ${extraClasses ?? ""}`}>
+        {links.map((link, index) => (
+          <FooterLink key={index} text={link.text} url={link.url} />
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+export default FooterMenu;
```
Comment (UmmarHamid) [inline app/components/FooterMenu/FooterMenu.tsx:25]:
```suggestion
    <div className={`${menuClasses} ${extraClasses ?? ""}`}>
      <Heading
        variant="20"
        text={title}
        extraClasses="text-white font-bold "
      />
      <ul >
        {links.map((link, index) => (
          <FooterLink key={index} text={link.text} url={link.url} />
        ))}
      </ul>
    </div>
  );
};

export default FooterMenu;
```

File: app/components/Footer/Footer.types.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { FooterMenuType } from "../FooterMenu/FooterMenu.types";
+
+export type FooterType = {
+  logo: {
+    imgSrc: string;
+    url: string;
+    altText: string;
+  };
+  menus: Array<FooterMenuType>;
+  extraClasses?: string;
```
Comment (UmmarHamid) [inline app/components/Footer/Footer.types.ts]:
```suggestion
  logo: LogoType;
  menus: Array<FooterMenuType>;
  extraClasses?: string;
```

File: app/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { FooterType } from "./Footer.types";
+import Image from "next/image";
+import classNames from "classnames";
+
+const Footer = ({ logo, menus, extraClasses }: FooterType) => {
+  let footerClasses = classNames(
+    "flex justify-around py-20 px-[260px] h-80 bg-darkBlue w-full"
+  );
+  return (
+    <footer className={`${footerClasses} ${extraClasses}`}>
```
Comment (UmmarHamid) [inline app/components/Footer/Footer.tsx]:
```suggestion
    <footer className={`${footerClasses} ${extraClasses ?? ''}`}>
```

File: app/layout.tsx [hunk 4]
```diff
@@ -1,22 +1,89 @@
...
+            altText: "Header Logo",
+          }}
+          menu={{
+            links: [
+              { text: "Movies", url: "/movies" },
+              { text: "Tv Shows", url: "/tvshows" },
+              { text: "People", url: "/people" },
+              { text: "More", url: "/more" },
+            ],
+          }}
```
Comment (UmmarHamid) [inline app/layout.tsx]: Can you kindly store these prop values in constant like you have done for footer, it will look better in the return section?

**Review by UmmarHamid — CHANGES_REQUESTED** [review]
Looks Awesome!

Small feedback

File: app/layout.tsx (hunk 4 above)
Comment (Saminsfw) [reply app/layout.tsx]: I have made the changes that you mentioned.

**Review by UmmarHamid — APPROVED** [review]

**Pull request merged**